app/controllers/AccountDetailsController.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import javax.inject.Inject
+
+import config.AppConfig
+import models.User
+import models.viewModels.AccountDetailsModel
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc._
+import services.EnrolmentsAuthService
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.Future
+
+class AccountDetailsController @Inject()(val messagesApi: MessagesApi,
+                                         val enrolmentsAuthService: EnrolmentsAuthService,
+                                         implicit val appConfig: AppConfig)
+  extends AuthorisedController with I18nSupport {
+
+  def accountDetails(): Action[AnyContent] = authorisedAction { implicit request =>
+    user =>
+      for {
```

Comment:
In scala, `for {...} yield {...}` is useful when handling multiple futures, but if you're only needing to use one then using a `.map` is probably more appropriate
E.g.
`handleAccountDetailsModel(user).map{accountModel => 
   Ok(views.html.account.accountDetails(user, accountModel)
}`